eidas_node/tests/test_xml.py

```diff
@@ -1,17 +1,22 @@
+import re
 from io import BytesIO
-from typing import BinaryIO, Optional, TextIO, cast
+from typing import BinaryIO, Optional, Set, TextIO, cast
 from unittest.mock import Mock, patch
 
 import xmlsec
 from django.test import SimpleTestCase
 from lxml.etree import Element, QName, SubElement
 
+from eidas_node.constants import XmlBlockCipher, XmlKeyTransport
 from eidas_node.errors import SecurityError
 from eidas_node.saml import EIDAS_NAMESPACES, Q_NAMES
 from eidas_node.tests.constants import CERT_FILE, DATA_DIR, KEY_FILE, NIA_CERT_FILE, SIGNATURE_OPTIONS, WRONG_KEY_FILE
-from eidas_node.xml import (XML_SIG_NAMESPACE, create_xml_uuid, decrypt_xml, dump_xml, get_element_path,
-                            is_xml_id_valid, parse_xml, remove_extra_xml_whitespace, remove_newlines_in_xml_text,
-                            sign_xml_node, verify_xml_signatures)
+from eidas_node.xml import (XML_SIG_NAMESPACE, create_xml_uuid, decrypt_xml, dump_xml, encrypt_xml_node,
+                            get_element_path, is_xml_id_valid, parse_xml, remove_extra_xml_whitespace,
+                            remove_newlines_in_xml_text, sign_xml_node, verify_xml_signatures)
+
+# This is an ugly hack but only for unit tests...
+LIBXML_VERSION = tuple(map(int, re.search(r'\((\d+)\.(\d+)\.(\d+)\)', xmlsec.__doc__).groups()))  # type: ignore
```

Comment:
Very ugly :(

Reply:
I hope we will get rid of it soon: https://trac.nic.cz/ticket/26906

Reply:
Can you think of a better way to get libxmlsec version or information about supported features?

Reply:
Probably not... Although it looks like the version is parsed on install time and does not reflect the subsequent library updates...

Reply:
Indeed. That's why it is useful only in a controlled environment such as CI.